### PR TITLE
fix(kit): hover state from MarkerIcon should only anchor elements

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/kit/components/marker-icon/marker-icon.component.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ import {TuiMarkerIconMode} from '@taiga-ui/kit/types';
 import {Observable} from 'rxjs';
 
 @Component({
-    selector: `tui-marker-icon`,
+    selector: `tui-marker-icon, [tuiMarkerIcon]`,
```
Comment (waterplea) [inline projects/kit/components/marker-icon/marker-icon.component.ts:20]: Let's make it more concrete `a[tuiMarkerIcon], button[tuiMarkerIcon]`

File: projects/kit/components/marker-icon/marker-icon.style.less [hunk 2]
```diff
@@ -81,18 +81,21 @@
...
         .transition(color);
-        cursor: pointer;
         color: var(--tui-link);
         background-color: var(--tui-secondary);
 
-        &:hover {
-            color: var(--tui-link-hover);
-            background-color: var(--tui-secondary-hover);
-        }
+        a& {
```
Comment (waterplea) [inline projects/kit/components/marker-icon/marker-icon.style.less:87]: It's common to use `button` too, let's add it.

**Review by waterplea — APPROVED** [review]
Minor comments

**Review by vladimirpotekhin — APPROVED** [review]

File: projects/kit/components/marker-icon/marker-icon.style.less (hunk 2 above)
Comment (splincode) [reply projects/kit/components/marker-icon/marker-icon.style.less]: fixed

File: projects/kit/components/marker-icon/marker-icon.component.ts (hunk 1 above)
Comment (splincode) [reply projects/kit/components/marker-icon/marker-icon.component.ts]: fixed

**Pull request merged**